File: server/models/Resource.js
const mongoose = require('mongoose');

const ResourceSchema = new mongoose.Schema({
    resourceName:{type:String, required:true},
    description:{type:String, required:true},
    quantity:{type:String, required:true},
    resident:{
        type:mongoose.Schema.Types.ObjectId,
        ref:'Resident',
        requied:true,
    },
    residentName:{type:String, required:true},
    apartmentCode:{type:String, required:true},
    status:{type:String, enum:['Active','Deleted'],default:'Active'},
    createdAt:{type:Date, default:Date.now},
});

module.exports = mongoose.model('Resource', ResourceSchema);

File: lib/models/resource.dart
class Resource {
  final String id;
  final String title;
  final String description;
  final String userId;
  final String userName;
  final String apartmentCode;

  Resource({
    required this.id,
    required this.title,
    required this.description,
    required this.userId,
    required this.userName,
    required this.apartmentCode,
  });

  factory Resource.fromJson(Map<String, dynamic> json) {
    return Resource(
      id: json['_id'],
      title: json['resourceName'],
      description: json['description'],
      userId: json['resident']['_id'] ?? json['resident'],
      userName: json['residentName'],
      apartmentCode: json['apartmentCode'],
    );
  }
}